x/dex/keeper/grpc_estimate_place_limit_order_test.go

```diff
@@ -0,0 +1,105 @@
+package keeper_test
+
+import (
+	"cosmossdk.io/math"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/neutron-org/neutron/v4/x/dex/types"
+)
+
+func (s *DexTestSuite) TestEstimatePlaceLimitOrderGTC() {
+	// GIVEN liquidity  A<>B
+	s.SetupMultiplePools(
+		NewPoolSetup("TokenA", "TokenB", 0, 1, 0, 1),
+		NewPoolSetup("TokenA", "TokenB", 0, 1, 1, 1),
+	)
+
+	// WHEN estimate GTC Limit order selling "TokenA"
+	resp, err := s.App.DexKeeper.EstimatePlaceLimitOrder(s.Ctx, &types.QueryEstimatePlaceLimitOrderRequest{
+		TokenIn:          "TokenA",
+		TokenOut:         "TokenB",
+		TickIndexInToOut: 4,
```

Comment:
it's out of scope of the pr, but `EstimatePlaceLimitOrder` looks outdated, TickIndexInToOut is not marked as deprecated and `limit_sell_price` is missed.
What do you think about to unify these quries with something like
```proto
message QueryEstimatePlaceLimitOrderRequest {
    MsgPlaceLimitOrder msg = 1;
}
```
to not copypaste whole bunch of fields. The only downside i can see you can get rid of `creator` and `receiver` in a query